lib/kb_Msuite/Utils/DataStagingUtils.py

```diff
@@ -81,6 +106,21 @@ def stage_input(self, input_ref, fasta_file_extension):
         return {'input_dir': input_dir, 'folder_suffix': suffix, 'all_seq_fasta': all_seq_fasta}
 
 
+    def count_fasta_seq_len(self, fasta_path):
+        '''
+        counts the number of non-header, non-whitespace characters in a FASTA file
+        '''
+        seq_len = 0
+        with open (fasta_path, 'r', 0) as fasta_handle:
+            for line in fasta_handle:
+                line = line.strip()
+                if line.startswith('>'):
+                    continue
+                line = line.replace(' ','')
+                seq_len += len(line)
```

Comment:
You don't have to go through the whole file. You can return as soon as it detects any valid sequence.

Reply:
Might want to apply a minimum threshold down the road.  This version permits that, and is fast enough when compared with the much slower steps in rest of App.  I do think about these things.

Reply:
you do make a good point though... I should abort after min_thresh is reached just on principle.  I'll fix that.

Reply:
modified method to have min_fasta_len threshold